nncf/quantization/algorithms/min_max/onnx_backend.py

```diff
@@ -125,49 +127,85 @@ def _get_input_edges_mapping(nncf_graph: NNCFGraph):
-            return None, use_abs_max
+    def _get_reduction_shape_for_weight(
+        node: NNCFNode, weight_port_id: int, is_per_channel: bool
+    ) -> Optional[ReductionShape]:
+        if not is_per_channel:  # Per-Tensor
+            return None
+        weight_shape = node.layer_attributes.weight_attrs[weight_port_id]["shape"]
+        reduction_shape = list(range(len(weight_shape)))
+        if len(reduction_shape) == 1:  # If only one channel
+            return tuple(reduction_shape)
+        axis = ONNXMinMaxAlgoBackend._get_weight_channel_axis(node, weight_port_id)
+        reduction_shape.pop(axis)
+        return tuple(reduction_shape)
 
-        if not target_point.is_weight_target_point():
-            # TODO: support reduction shapes for 3D-5D conv cases
-            return (0, 2, 3), use_abs_max
+    @staticmethod
+    def _get_reduction_shape_for_activation(
+        nncf_graph: NNCFGraph, node: NNCFNode, target_point, is_per_channel: bool
+    ) -> Optional[ReductionShape]:
+        if not is_per_channel:  # Per-Tensor
+            return None
 
-        # Calculate reduction shape for weight statistic collector
-        node = nncf_graph.get_node_by_name(target_point.target_node_name)
-        assert node.layer_attributes.has_weight()
-        weight_shape = node.layer_attributes.weight_attrs[target_point.port_id]["shape"]
-        reduction_shape = list(range(len(weight_shape)))
+        if target_point.type == TargetType.PRE_LAYER_OPERATION:
+            shape = nncf_graph.get_input_edges(node)[target_point.port_id].tensor_shape
+        elif target_point.type == TargetType.POST_LAYER_OPERATION:
+            shape = nncf_graph.get_output_edges(node)[target_point.port_id].tensor_shape
+        else:
+            raise NotImplementedError(f"Unsupported target point type {target_point.type}.")
+        if not shape:  # ONNX model can not have a shape of a edge, even after shape inference.
+            if target_point.type == TargetType.PRE_LAYER_OPERATION:
+                nncf_logger.info(
+                    f"The shape of input edge of a node {node.node_name} is unkown. Therefore per-tensor quantizaiton is applied."
+                )
+            elif target_point.type == TargetType.POST_LAYER_OPERATION:
+                nncf_logger.info(
+                    f"The shape of output edge of a node {node.node_name} is unkown. Therefore per-tensor quantizaiton is applied."
+                )
+            nncf_logger.info(f"Please consider to run pre-processing before quantization.")
+            # TODO: add preprocessing tool for ONNX model.
+            return None
 
-        axis = ONNXMinMaxAlgoBackend._get_axis(nncf_graph, target_point, quantizer_config)
-        reduction_shape.pop(axis)
-        return tuple(reduction_shape), use_abs_max
+        # TODO (l-bat): Disable quantizer propogation through layout changing operations
+        channel_axis = 1  # Activations have channel first layout: [N, C, Z, Y, X]
```

Comment:
Is it correct for `PRE_LAYER_OPERATION` and `POST_LAYER_OPERATION`? For example, for `MatMul` layers.
Why don't we use the `output_channel_axis` attribute, for example?

Reply:
This is a copy past from OV - https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/min_max/openvino_backend.py#L142